src/box/lua/config/applier/console.lua

```diff
@@ -10,34 +14,49 @@ local function socket_file_to_listen_uri(file)
     return ('unix/:./%s'):format(file)
 end
 
+local function close_listening_socket()
+    if listening_socket ~= nil then
+        listening_socket:close()
+        listening_socket = nil
+        listening_socket_file = nil
+    end
+end
+
 local function apply(config)
     local configdata = config._configdata
     local enabled = configdata:get('console.enabled', {use_default = true})
     if not enabled then
         log.debug('console.apply: console is disabled by the ' ..
             'console.enabled option; skipping...')
+        -- If the console is disabled, we must close the active socket.
+        close_listening_socket()
         return
     end
 
-    local socket_file = configdata:get('console.socket', {use_default = true})
-    assert(socket_file ~= nil)
+    local new_socket_file = configdata:get(
+        'console.socket', {use_default = true}
+    )
+    assert(new_socket_file ~= nil)
 
     -- The same socket file is pointed by different paths before
     -- and after first box.cfg() if the `process.work_dir` option
     -- is set.
     local iconfig_def = configdata._iconfig_def
-    socket_file = instance_config:prepare_file_path(iconfig_def, socket_file)
-
-    local listen_uri = socket_file_to_listen_uri(socket_file)
-    log.debug('console.apply: %s', listen_uri)
-
-    -- Ignore 'Address already in use' error, because it is
-    -- natural effect of reloading the configuration. All others
-    -- errors are re-raised and lead to failure of apply of the
-    -- configuration.
-    local ok, res = pcall(console.listen, listen_uri)
-    if not ok and errno() ~= errno.EADDRINUSE then
-        error(res)
+    new_socket_file = instance_config:prepare_file_path(
+        iconfig_def, new_socket_file)
+    local new_socket_file_path = fio.abspath(new_socket_file)
+
+    -- If console.socket has changed we must close the listening socket.
+    if listening_socket_file ~= new_socket_file_path and
+            listening_socket ~= nil then
```

Comment:
The `listening_socket ~= nil` check is already in `close_listening_socket()`, it is redundant here.

Reply:
extra check removed